src/droplet.F

```diff
@@ -515,35 +550,61 @@ subroutine droplet_driver(dt,xh,uh,ruh,xf,yh,vh,rvh,yf,zh,mh,rmh,zf,mf,zs,    &
 
       if( terrain_flag )then
             call getparcelzs(xh,uh,ruh,xf,yh,vh,rvh,yf,zs,pdata)
-            DO np=1,nparcelsActive
+            DO np=1,nparcelsLocalActive
               ! get z from sigma:
               ! (see Section 3 of "The governing equations for CM1", 
               !  http://www2.mmm.ucar.edu/people/bryan/cm1/cm1_equations.pdf)
               pdata(np,prz) = pdata(np,przs) + pdata(np,prsig)*((zt-pdata(np,przs))*rzt)
             ENDDO
       endif
+
     if(timestats.ge.1) time_droplet=time_droplet+mytime()
 
-!
 !----------------------------------------------------------------------
+! Final thing: Inject droplets which will be active on the next call of droplet_driver
+! Requires inputting the current 10-meter wind speed
+   
+   if ( .not. use_wall_bc ) then  
+      s10_xym = 0.0
+      do j=1,nj
+         do i=1,ni
+            s10_xym = s10_xym + s10(i,j)
+         end do
+      end do
+#ifdef MPI
+      call mpi_allreduce(mpi_in_place,s10_xym,1,mpi_real,mpi_sum,mpi_comm_world,ierr)
+#endif
+      s10_xym = s10_xym/real(nx*ny)
+      if (myid==0) write(*,*) 'DHR U10',s10_xym
+      call droplet_injection(pdata,xf,yf,dt,s10_xym,randomNumbers)
+#ifdef MPI
+      call mpi_allreduce(nparcelsLocalActive,np_tmp,1,mpi_int,mpi_sum,mpi_comm_world,ierr)
+#endif
+      if (myid==0) then 
+         write(*,'(a8,3i)')  'DHR1:',myid,np_tmp,nparcelsLocalActive
+      end if
```

Comment:
Do we need to keep this print statement?

Reply:
No, not in the long run. I'll remove it for the next PR